tools/BuildPackagesTask/Microsoft.Azure.Build.Tasks/SetGenerator.cs

```diff
@@ -152,5 +120,37 @@ public static HashSet<string> Generate(HashSet<string> filesChangedSet, Dictiona
 
             return outputSet;
         }
+        /// <summary>
+        /// Generate set according to module names
+        /// </summary>
+        /// <param name="moduleNames"></param>
+        /// <param name="mappingsDictionary"></param>
+        /// <returns></returns>
+        public static HashSet<string> Generate(string[] moduleNames, Dictionary<string, string[]> mappingsDictionary)
+        {
+            var outputSet = new HashSet<string>();
+
+            foreach (var module in moduleNames)
+            {
+                foreach (var pair in mappingsDictionary)
```

Comment:
Suggest to use mappingDictionary.ContainsKey($"src/{module}/")

Reply:
Updated.